rust/connlib/snownet/src/node.rs

```diff
@@ -841,6 +841,12 @@ where
                     .values_mut()
                     .find(|a| a.server().matches(from))
                 else {
+                    if crate::channel_data::decode(packet).is_ok() {
+                        tracing::debug!("Packet was a channel data message for unknown allocation");
```

Comment:
Any chance this opens us up for DoS?

Reply:
What scenario are you thinking of in particular?

The intent here is to only change _where_ the packet is discarded and what level that is logged at.

Previously, this would fall through to the very end after trying all connections. Now we special-case it here by checking if it is a valid channel-data packet (which automatically means it is not a valid WireGuard packet).

Reply:
We can maybe make this clearer by enforcing it to parse as a valid channel-data packet first.